modules/fedimint-ln-client/src/db.rs

```diff
@@ -48,7 +48,7 @@ pub struct PaymentResultPrefix;
 #[derive(Debug, Encodable, Decodable, Serialize)]
 pub struct PaymentResult {
     pub index: u16,
-    pub completed_payment: Option<(PayType, ContractId)>,
+    pub completed_payment: Option<(PayType, ContractId, Amount)>,
```

Comment:
What's the idea here? Is this a precursor to upping the fee on next try?

Reply:
No, since we now return the fee the gateway takes back to the client, we now just need to remember it if `pay_bolt11_invoice` is ever called on the same invoice for an already completed payment.

I made a new struct called `OutgoingLightningPayment` to capture these fields.